clean-data/README.md

```diff
@@ -0,0 +1,58 @@
+# Cleaning and Combine Free Code Camp Survey Data
+
+## Introduction
+
+The survey data was broken up into two parts and need to be combined into one
+for ease of future downstream analyses. Additionally, these two data sets need
+to be cleaned up a bit because of the nature of survey data.
+
+## Notable Data Transformations
+
+### Obvious Outliers
+
+In some of the numeric free text answers, numeric values were filtered out if it
+was beyond a reasonable threshold. For example, an answer saying you've coded
+for 100,000 months would be removed.
+
+### Numeric Ranges
+
+Some answers were given as ranges. For example, a range of "9-10" months of
+programming might have been answer to a question. The average of this range was
+taken when possible.
+
+### Years to Months
+
+Some answers to a question asking about months were given in years. These were
+converted to months if possible.
+
+### Normalization of Answers
+
+Some of the free text answers were very similar to each other, with the
+exception of a space or two. These will register as different answers if you
+aren't looking for them. Answers like "Cybersecurity" and "Cyber Security" are
+the same and were changed to a consistent manner. There may have been some
+missed.
+
+
+## Prerequisites to Rerun Data Manipulations
+
+- [R](https://www.r-project.org/) (>= 3.2.3)
+- [dplyr](https://github.com/hadley/dplyr) (>= 0.4.3)
+
```

Comment:
The script is still a work in progress. It won't completely clean the data after you run the script. You have to run the script interactively line-by-line (or function-by-function). I am using RStudio and it makes it easier to do this. Most of the functions will work up until cleaning the second dataset. 

I would run the following code after you've read those functions into R, while being in the `clean-data/` directory and renamed the original data directory to `raw-data/`.

```r
dat <- read_in_data()
part1 <- rename_part_1(dat$part1)
part2 <- rename_part_2(dat$part2)
```

I hope that helps.